builder/azure/arm/config.go

```diff
@@ -86,6 +86,9 @@ type SharedImageGallery struct {
 	// version across regions set this value to one that is available in all
 	// regions where you are deploying.
 	ImageVersion string `mapstructure:"image_version" required:"false"`
+
+	// Id of the community gallery image : /CommunityGalleries/{cgName}/Images/{img}[/Versions/{}] (Versions part is optional)
+	CommunityGalleryImageId string `mapstructure:"communityGallery_image_id" required:"false"`
```

Comment:
I'm also adding Community Gallery support for CAPZ: https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2277, so couple of comments from experience from there:
- If community gallery is specified, then neither `Subscription` and `ResourceGroup` don't need to be specified, so perhaps they should be made optional + some logic be added to validate the configuration.
- With the above, image ID can be computed using already provided fields.

Reply:
maybe we should just mimic the VM api, the gallery name is also a bit different. For e.g., for community galleries, it is the 'public' nme of the gallery, not the gallery name used by the publisher of community gallery. That's why I decided to keep them separate

Reply:
I agree we should keep them consistent across the clients!